src/Reconcile.js

```diff
@@ -54,83 +42,106 @@ export function resolve (fiber, host, parent, a, b, c, type) {
  * @param {object} fiber
  * @param {object} host
  * @param {object} parent
- * @param {object} a
- * @param {object} b
- * @param {object} c
- * @param {number} idx
+ * @param {object} element
+ * @param {object} snapshot
+ * @param {(object|object[])} value
+ * @param {object[]} a
+ * @param {object[]} b
  */
-export function update (fiber, host, parent, a, b, c, idx) {
-	if (a.type === b.type) {
-		switch (a.uid) {
-			case Enum.text:
-				if (a.children !== b.children) {
-					Schedule.dispatch(fiber, Enum.content, host, parent, a, a.children = b.children)
-				}
-			case Enum.empty:
-				return
-			case Enum.component:
-				return Schedule.dispatch(fiber, Enum.component, host, a, b, {})
-			case Enum.thenable:
-				return resolve(fiber, host, parent, a, b, a.children, a.type = b.type)
-		}
-
-		children(fiber, host, a, a.children, b.children)
-		props(fiber, host, parent, a, object(a.props, a.props = b.props))
-	} else {
-		if (a.uid === b.uid) {
-			switch (a.type = b.type, a.uid) {
-				case Enum.target:
-					Schedule.dispatch(fiber, Enum.mount, host, Node.reparent(a, object(a.props, a.props = {})), a, a)
-				case Enum.thenable:
-					return update(fiber, host, parent, a, b, c, idx)
-			}
-		}
-
-		replace(fiber, host, parent, a, b, c, idx)
+export function enqueue (fiber, host, parent, element, snapshot, value, a, b) {
+	if (a.parent !== null) {
+		children(fiber, host, parent, 0, a, a === b ? Element.resolve(type, snapshot.props) : value)
 	}
 }
 
 /**
  * @param {object} fiber
  * @param {object} host
  * @param {object} parent
- * @param {object} a
- * @param {object} b
- * @param {object} c
- * @param {number} idx
+ * @param {object} element
+ * @param {object} snapshot
+ * @param {object[]} siblings
+ * @param {number} index
  */
-export function replace (fiber, host, parent, a, b, c, idx) {
-	Schedule.dispatch(fiber, Enum.mount, host, parent, Node.create(fiber, host, parent, c[idx] = b), a)
-	Schedule.dispatch(fiber, Enum.unmount, host, parent, a, Node.destroy(fiber, a))
+export function replace (fiber, host, parent, element, snapshot, siblings, index) {
+	Schedule.dispatch(fiber, Enum.mount, host, parent, Node.create(fiber, host, parent, siblings[index] = snapshot, null), element)
+	Schedule.dispatch(fiber, Enum.unmount, host, parent, element, Node.destroy(fiber, element))
 }
 
 /**
  * @param {object} fiber
  * @param {object} host
  * @param {object} parent
- * @param {object} a
- * @param {object} b
+ * @param {object} element
+ * @param {object} snapshot
+ * @param {object[]} siblings
+ * @param {number} index
  */
-export function props (fiber, host, parent, a, b) {
-	if (b) {
-		Schedule.dispatch(fiber, Enum.props, host, parent, a, b)
+export function update (fiber, host, parent, element, snapshot, siblings, index) {
+	// ***** DO MIND ME *****
+	// TODO uncomment once the setTimeout perf cliff is figured out
```

Comment:
the only reason I see `setTimeout` causing a performance problem is if the scheduler is awaiting timeouts serially. (guessing)

Reply:
I'm starting to think setTimout is just not that fast when compared side-by-side to optimised JavaScript in the context of the contrived benchmark, given it's a browser API involving some level of i/o.